src/plugins/intel_npu/src/plugin/npuw/partitioning/online/snapshot.cpp

```diff
@@ -664,6 +1008,26 @@ void Snapshot::completeRepeating(const std::shared_ptr<Repeated>& reptag, const
     m_layer_matches.insert({tag, layer_matches});
 }
 
+GPtrSet Snapshot::getRepGroups(const Group::GPtr& group) const {
+    auto rep = group->repeated();
+
+    GPtrSet repeating_groups;
+
+    for (const auto& nh_other : m_graph->sorted()) {
```

Comment:
I don't think you need to walk over a sorted graph here - and pay the price of the topologic sort all the time. Just go over the list of nodes.

Reply:
Done